src/qdl.js

```diff
@@ -283,7 +283,7 @@ export class qdlDevice {
       if (offset % gpt.sectorSize !== 0) {
         throw "qdl - Offset not aligned to sector size";
       }
-      const sector = (partition.start + BigInt(offset)) / BigInt(gpt.sectorSize);
+      const sector = partition.start + BigInt(offset / gpt.sectorSize);
```

Comment:
There was a typo in the calculation for the sector to flash the chunk to, causing the wrong sectors to be overwritten